### PR TITLE
operator: Promote AlertingRules, RecordingRules and RulerConfig from v1beta1 to v1

### PR DESCRIPTION
**Pull request by aminesnow**
**What this PR does / why we need it**:
This PR promotes the `AlertingRules`, `RecordingRules` and `RulerConfig` CRDs from `v1beta1` to `v1`.
A conversion `webhook` has been added.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by Red-GV — COMMENTED** [review]
Think we also need to regenerate the docs for the api. Otherwise, lgtm.

**Review by periklis — COMMENTED** [review]
LGTM

File: operator/apis/loki/v1beta1/alertingrule_types.go [hunk 1]
```diff
@@ -105,9 +107,8 @@ type AlertingRuleStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
-//+kubebuilder:object:root=true
-//+kubebuilder:subresource:status
-//+kubebuilder:webhook:path=/validate-loki-grafana-com-v1beta1-alertingrule,mutating=false,failurePolicy=fail,sideEffects=None,groups=loki.grafana.com,resources=alertingrules,verbs=create;update,versions=v1beta1,name=valertingrule.loki.grafana.com,admissionReviewVersions=v1
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
```
Comment (periklis) [inline operator/apis/loki/v1beta1/alertingrule_types.go:112]:
```suggestion
// +kubebuilder:object:root=true
// +kubebuilder:subresource:status
//+kubebuilder:unservedversion
```

File: operator/apis/loki/v1beta1/recordingrule_types.go [hunk 2]
```diff
@@ -85,7 +87,6 @@ type RecordingRuleStatus struct {
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
```
Comment (periklis) [inline operator/apis/loki/v1beta1/recordingrule_types.go:89]:
```suggestion
//+kubebuilder:object:root=true
//+kubebuilder:subresource:status
//+kubebuilder:unservedversion
```

File: operator/apis/loki/v1beta1/rulerconfig_types.go [hunk 3]
```diff
@@ -522,7 +524,6 @@ type RulerConfigStatus struct {
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
```
Comment (periklis) [inline operator/apis/loki/v1beta1/rulerconfig_types.go:526]:
```suggestion
//+kubebuilder:object:root=true
//+kubebuilder:subresource:status
//+kubebuilder:unservedversion
```

**Review by periklis — CHANGES_REQUESTED** [review]

**Review by periklis — APPROVED** [review]

**Pull request merged**